### PR TITLE
Feat: Flutter UI Components

### PR DESCRIPTION
**Pull request by rachana-egov**

### CODE REVIEW

File: flutter/digit-ui-components/digit_components/lib/digit_components.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+library digit_components;
+
+// import 'package:google_fonts/google_fonts.dart';
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/digit_components.dart:3]: remove un-used imports

File: flutter/digit-ui-components/digit_components/lib/digit_components.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export 'widgets/widgets.dart';
+
+class DigitUi {
+  static const DigitUi _instance = DigitUi._internal();
+  static DigitUi get instance => _instance;
+
+  const DigitUi._internal();
+
+  Future<void> initThemeComponents() async {
+    // GoogleFonts.config.allowRuntimeFetching = false;
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/digit_components.dart:15]: remove un used code

File: flutter/digit-ui-components/digit_components/lib/models/chipModel.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  final Widget? separator;
+
+  final bool autoScroll;
+
+  const ChipConfig({
+    this.deleteIcon,
+    this.deleteIconColor = Colors.white,
+    this.backgroundColor,
+    this.padding = const EdgeInsets.only(left: 12, top: 0, right: 4, bottom: 0),
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/models/chipModel.dart:26]: use kpadding 
and move constant values to constants folder

File: flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_checkbox.dart [hunk 4]
```diff
@@ -0,0 +1,115 @@
+
+import 'package:digit_components/digit_components.dart';
+import 'package:digit_components/widgets/atoms/digit_checkbox_icon.dart';
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+import '../../constants/app_constants.dart';
+import '../../enum/app_enums.dart';
+
+/// `DigitCheckbox` is a checkbox component with a label and optional icon customization.
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/widgets/atoms/digit_checkbox.dart:9]: change the comment pattern

File: flutter/digit-ui-components/digit_components/lib/widgets/widgets.dart [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
+export './powered_by_digit.dart';
+export './scrollable_content.dart';
+
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/lib/widgets/widgets.dart:4]: remove extra line space

File: flutter/digit-ui-components/digit_components/test/digit_ui_components_test.dart [hunk 6]
```diff
@@ -0,0 +1 @@
+void main() {}
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/test/digit_ui_components_test.dart:1]: why empty method

File: flutter/digit-ui-components/digit_components/test/theme/theme_test.dart [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                Text('Headline XL', style: theme.displayMedium),
+                Text('Headline L', style: theme.headlineLarge),
+                Text('Headline M', style: theme.headlineMedium),
+                Text('Headline S', style: theme.headlineSmall),
+                Text('Caption XL', style: theme.labelLarge),
+                Text('Caption L', style: theme.labelMedium),
+                Text('Caption M', style: theme.labelSmall),
+                Text('Body L', style: theme.bodyLarge),
+                Text('Body S', style: theme.bodyMedium),
+                Text('Body XS', style: theme.bodySmall),
```
Comment (naveen-egov) [inline flutter/digit-ui-components/digit_components/test/theme/theme_test.dart:29]: 1 more type is missing pls confirm with UI/Ux team

**Review by naveen-egov — CHANGES_REQUESTED** [review]
Added code comments pls check and re-submit

File: flutter/digit-ui-components/digit_components/test/theme/theme_test.dart (hunk 7 above)
Comment (rachana-egov) [reply flutter/digit-ui-components/digit_components/test/theme/theme_test.dart:29]: will do it, once i get the confirmation

**Review by naveen-egov — APPROVED** [review]

**Pull request merged**